### PR TITLE
Caching writer

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] cleaner tests
- [x] hook up with BCW
- [x] extract constants 
- [x] needs more tests
  - [x] after first build, js or css change re-generates SRI accordingly in index accordingly (needs to pass)
  - [x] test that it propagates ALL files forward.
- [x] pending a BCW bug I think i found ->  https://github.com/ember-cli/broccoli-caching-writer/issues/64

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
   this.options = options || {};
   this.context = this.options.context || {};
-  Filter.call(this, inputNode);
+  var nodes = inputNodes;
+  if (!Array.isArray(nodes)) {
+    nodes = [nodes];
+  }
+
+  CachingWriter.call(this, nodes, {
+    cacheInclude: [
```
Comment (stefanpenner) [inline index.js]: are only js and css files taken into account, or are all assets?

Comment (rwjblue) [inline index.js]: The regexps only look for `<script>` and `<link>` elements.

Comment (stefanpenner) [inline index.js]: ah ok

**Comment by rwjblue** [comment]
Looking good so far!

**Comment by stefanpenner** [comment]
I realized, there is 1 important thing this will need to do. Propagate all otherFiles, even if the cache wasn't busted. 

An example would be.
`foo.png` changes, although it doesn't trigger a build, we must still copy it. For this, I will need to add a small feature to BCW. Likely an additional hook, I need to think what that might be.

**Comment by stefanpenner** [comment]
I haven't  had any more time tonight, and tomorrow will be quite pressed with other tasks. Simplest thing i see if to opt out of caching and land this. @rwjblue when you wake up, let me know if you think of something else. I suspect.

**Comment by rwjblue** [comment]
Latest update looks good to me.

**Comment by stefanpenner** [comment]
Its worth noting, working on this gave me an idea to improve ergonomics of writing more advanced broccoli plugins. BCW and BF kinda feel awkward in all by the simplest scenarios.

**Comment by jonathanKingston** [comment]
Thanks so much for all your hard work on @stefanpenner. Totally agree on the ergonomics point.
I'll look at snyk and ESLint at some point.

**Pull request merged**